.github/workflows/python-package.yml

```diff
@@ -43,3 +40,13 @@ jobs:
       run: |
         # run tests in tests/ dir and only fail if there are failures or errors
         pytest tests/ --verbose --failed-first --exitfirst --disable-warnings
+      - run: black --check .
+      - name: Commit black changes, if any
+        if: failure()
+        run: |
+          black .
+          git config --global user.name github-actions
+          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'          
```

Comment:
So a bot will force black the changes?

Reply:
it looks like it will. My good friend ChatGPT helped out here, but also found a template here. https://github.com/badguy99/octoblock/blob/master/.github/workflows/black.yml

Seems like we need a secret for the bot to have access and push to the branch